src/Core/3DTiles/C3DTBatchTable.js

```diff
@@ -17,41 +18,57 @@ import C3DTilesTypes from './C3DTilesTypes';
  */
 class C3DTBatchTable {
     /**
-     * @param {ArrayBuffer} buffer - batch table buffer to parse.
-     * @param {ArrayBuffer} binaryLength - the length of the binary part of
-     * the batch table (not supported yet)
+     * @param {ArrayBuffer} buffer - batch table buffer to parse
+     * @param {number} jsonLength - batch table json part length
+    * @param {number} binaryLength - batch table binary part length
      * @param {number} batchLength - the length of the batch.
      * @param {Object} registeredExtensions - extensions registered to the layer
      */
-    constructor(buffer, binaryLength, batchLength, registeredExtensions) {
+    constructor(buffer, jsonLength, binaryLength, batchLength, registeredExtensions) {
         this.type = C3DTilesTypes.batchtable;
         this.batchLength = batchLength;
```

Comment:
I think it's a good idea to add this test to check if the provided arguments seem correct but not to check if the constructor was called with the former signature since the test can pass if the constructor is called with the former signature and if the batch length equals the binary length.

However, since there is no optional argument in this function, we can test if the number of arguments is correc and add a deprecation warning and compute the values as suggested if not. WDYT?

Reply:
Indeed, I didn't think of simply counting the number of arguments but it is a safer test :+1:

Reply:
After some more thinking, I think we should not add a deprecation test based on arguments length either since it may happen for other reasons than because the previous constructor was used. In addition, batch tables are not meant to be constructed directly by users of itowns but by our 3D tiles parsers so there is little chance that a user calls this constructor anyway. Therefore, I think we should keep it this way and only add BREAKING CHANGE in the commit title (and therefore in the release). WDYT ?